File: app/src/main/java/kr/hs/emirim/sookhee/redonorpets/adapter/CommentAdapter.java
package kr.hs.emirim.sookhee.redonorpets.adapter;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.squareup.picasso.Picasso;

import java.util.HashMap;

import kr.hs.emirim.sookhee.redonorpets.R;
import kr.hs.emirim.sookhee.redonorpets.model.CommentData;

public class CommentAdapter extends RecyclerView.Adapter<CommentAdapter.CustomViewHolder> {

    private Context mCtx;
    private HashMap<String, CommentData> mData;

    public CommentAdapter(Context mCtx) {
        this.mCtx = mCtx;
        mData = new HashMap<>();
    }

    @NonNull
    @Override
    public CustomViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View v = LayoutInflater.from(mCtx).inflate(R.layout.comment_item, parent, false);

        return new CustomViewHolder(v);
    }

    @Override
    public void onBindViewHolder(@NonNull CustomViewHolder holder, final int position) {
        CommentData chat = (CommentData) mData.values().toArray()[position];

        String img = chat.getImg();

        holder.tvName.setText(chat.getName());
        holder.tvContent.setText(chat.getContent());
        Picasso.get().load(img).into(holder.ivProfile);

    }

    @Override
    public int getItemCount() {
        return mData.size();
    }

    class CustomViewHolder extends RecyclerView.ViewHolder{

        TextView tvName;
        TextView tvContent;
        ImageView ivProfile;
        public View pView;

        public CustomViewHolder(View itemView) {
            super(itemView);

            this.pView = itemView;
            tvName = itemView.findViewById(R.id.commentNameTextView);
            tvContent = itemView.findViewById(R.id.commentContentTextView);
            ivProfile = itemView.findViewById(R.id.commentProfileImageView);

        }

    }


    public void addDataAndUpdate(String key, CommentData p){
        mData.put(key, p);
        notifyDataSetChanged();
    }

    public void deleteDataAndUpdate(String key){
        mData.remove(key);
        notifyDataSetChanged();
    }

    public void clearData(){
        mData.clear();
        notifyDataSetChanged();
    }

}

File: app/src/main/java/kr/hs/emirim/sookhee/redonorpets/ui/story/CommentAdapter.kt
package kr.hs.emirim.sookhee.redonorpets.ui.story

import android.annotation.SuppressLint
import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import kr.hs.emirim.sookhee.redonorpets.common.setImageWithUrl
import kr.hs.emirim.sookhee.redonorpets.databinding.ItemCommentBinding
import kr.hs.emirim.sookhee.redonorpets.domain.entity.Comment

/**
 *  CommentAdapter.kt
 *
 *  Created by Minji Jeong on 2022/01/12
 *  Copyright © 2022 DonorPets2.0. All rights reserved.
 */

class CommentAdapter :
    RecyclerView.Adapter<CommentAdapter.ViewHolder>() {

    private var items: List<Comment> = emptyList()

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val inflater = LayoutInflater.from(parent.context)
        return ViewHolder(ItemCommentBinding.inflate(inflater, parent, false))
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        holder.bind(items[position])
    }

    override fun getItemCount(): Int {
        return items.size
    }

    @SuppressLint("NotifyDataSetChanged")
    fun setItem(list: List<Comment>) {
        items = list
        notifyDataSetChanged()
    }

    class ViewHolder(
        private val binding: ItemCommentBinding,
    ) : RecyclerView.ViewHolder(binding.root) {

        fun bind(comment: Comment, ) {
            binding.comment = comment
            binding.ivProfile.setImageWithUrl(comment.profileImage)
        }
    }
}
